src/screens/searchResult/screen/tabs/communities/container/communitiesResultsContainer.js

```diff
@@ -48,25 +49,26 @@ const CommunitiesResultsContainer = ({ children, navigation, searchValue }) => {
     getCommunities('', searchValue ? 100 : 20, searchValue || null, 'rank')
       .then((communities) => {
         if (currentAccount && currentAccount.username) {
-          getSubscriptions(currentAccount.username).then((subs) => {
-            if (subs) {
-              communities.forEach((community) =>
-                Object.assign(community, {
-                  isSubscribed: subs.some(
-                    (subscribedCommunity) => subscribedCommunity[0] === community.name,
-                  ),
-                }),
-              );
-            }
-            if (searchValue) {
-              setData(communities);
-            } else {
-              setData(shuffle(communities));
-            }
-            if (communities.length === 0) {
-              setNoResult(true);
-            }
-          });
+          if (subscribedCommunities.data && subscribedCommunities.data.length) {
```

Comment:
apparently this section is not considering `subscribedCommunitiesCache` as done is communityContainer, perhaps it can also use similar approach...

https://github.com/ecency/ecency-mobile/blob/e328e0c545a32e6042218162af8f10971991eae2/src/screens/community/container/communityContainer.js#L56

Check how Gaming Hive group is actually joined but search is still showing option to join communities while is should be showing leave, I did try refreshing the list but since data it is not yet updated in `subscribedCommunities` store and only available in `subscribedCommunitiesCache` which is not being considered here.
https://user-images.githubusercontent.com/6298342/182150778-ed235f21-c0af-41b8-ba65-7db754904d58.mov